modules/doc/content/ncrc/applications/ncrc_develop.md.template

```diff
@@ -0,0 +1,159 @@
+
+- +Personal Machine (Conda)+:
+
+  ```bash
+  mamba create -n moose moose-tools moose-libmesh
+  ```
+
+- +[!ac](INL) [!ac](HPC) Sawtooth or Lemhi+:
+
+  It will be necessary to build PETSc and libMesh using the supplied build scripts in
+  `moose/scripts`. The time it takes to build both PETSc and libMesh can be hours:
+
+  ```bash
+  module load use.moose moose-dev    # a reminder to have these loaded
+  export MOOSE_JOBS=6 METHODS=opt    # a reminder to set these
+  cd ~/projects/{{binary}}/moose
+  scripts/update_and_rebuild_petsc.sh
+  scripts/update_and_rebuild_libmesh.sh
+  ```
+
+  !alert tip title=MOOSE_JOBS? METHODS?
+  Go to [MOOSE Build System](application_development/build_system.md) for more information on
+  available methods. MOOSE_JOBS controls how many cores to use during the use of our scripts.
+
+## Build {{ApplicationName}} id=build
+
+Building {{ApplicationName}} starts with entering the directory, bringing in any updates from our
+upstream remote and finally running `make`. Before we begin, +burn the next two warnings into your
+brain+:
+
+!alert warning title=You performed a submodule update?
+`Then you should perform a Conda update` (if you are using Conda).
+
+!alert warning title=You updated Conda?
+`Then you should perform a submodule update`.
+
+The symmetry importance comes from our Conda packages being version controlled by and within the
+MOOSE repository. So any updates to MOOSE (the `{{binary}}/moose` submodule) requires you to stay on
+top of any Conda updates, or rebuilding PETSc and libMesh.
+
+If this is your first time following these instructions there should be no updates (you have a brand
+new fork/clone/conda install). We will perform an update to familiarize yourself with the process:
+
+```bash
+cd ~/projects/{{binary}}
+git fetch upstream
+git branch   # take note of the results
+```
+
+`git fetch upstream` updates your local references with that of the remote's. It does not 'modify'
+any source files. It will return nothing if there was nothing new to fetch on the `upstream` remote.
+
+`git branch` reports what branch you have checked out. It is impossible to write a document for
+every application's branch naming schemes, so whatever was reported with `git branch`, use that for
+the next command.
+
+Next, we will perform a `rebase`. This will make changes to your repository, if there are changes to
+be made:
+
+```bash
+git rebase -i upstream/branch   # replace 'branch' with the results from `git branch`
+```
+
+If there were errors during the rebase process, that means you have changes which either need
+committing, or conflicts which need addressing.
+
+The {{binary}} repository is now up to date, and understands how to update it's submodules. Do that
+now:
+
+```bash
+git submodule update --init
+```
+
+If you whitness any updates to `{{binary}}/moose`, now is the time to apply those two earlier
+warnings about also needing to update your Conda packages or rebuilding PETSc and libMesh:
```

Comment:
- whitness -> witness. But "observe" is probably a better word here.
- Earlier in your warnings, you made it sound like you should always update the other, but here, there is a conditional (if you witness). You should probably make this consistent - probably amend the earlier warning.

Reply:
I realize there are times an update or rebuild is unnecessary, but I am trying to sear this notion into their development process;

- If you update your submodules
  - you should update conda
- if you update conda packages
  - your should update your submodules  

I realize this isn't always 100% true. But if they do it anyway, they will never have an issue... so 🤷 why not.

Reply:
That's fine, just wanted to make them consistent, which you have 👍